src/serialization/generic_record/FieldKind.ts

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * FieldKind for Compact.
+ * It is designed to be used with {@link GenericRecord.getFieldKind} API.
+ *
+ * Note that actual ids in {@link FieldType} and {@link FieldKind} are not matching.
+ * {@link FieldType} is the old API for Portable only and only meant to be used with
+ * {@link PortableReader.getFieldType} API.
+ *
+ * This API is currently in Beta and can change at any time.
+ */
+export enum FieldKind {
```

Comment:
I think you mean me putting it under the generic_record folder. Ok we can move it to serialization folder. In Java they are next to each other in serialization package. There is not generic_record folder

Reply:
> Also... you are not specifying values here - except eg for INT16 - wouldn't it be safer to specify all values?

Ok I agree it will be safer doing it now.